src/csvcubed/writers/helpers/qbwriter/dsdtordfmodelshelper.py

```diff
@@ -301,17 +301,29 @@ def _get_qb_component_specs_for_col(
                 self._get_qb_attribute_specification(column_name_uri_safe, component)
             ]
         elif isinstance(component, QbMultiUnits):
-            return [self._get_qb_units_column_specification(column_name_uri_safe)]
+            return self._get_qb_units_column_specification(column_name_uri_safe)
         elif isinstance(component, QbMultiMeasureDimension):
             return self._get_qb_measure_dimension_specifications(component)
         elif isinstance(component, QbObservationValue):
             return self._get_qb_obs_val_specifications(component)
         else:
             raise TypeError(f"Unhandled component type {type(component)}")
 
+    _units_component_already_defined: bool = False
```

Comment:
Could you move this up to the top of the class where the other attributes are defined. Can you also declare that this isn't an attribute which should appear in the `__init__` function? The dev using this class shouldn't be allowed to configure this themselves.

Reply:
sorted